src/lib/ipfs.js

```diff
@@ -59,30 +64,98 @@ async function startIPFS () {
-            fs.writeFileSync(`${pathStore}${file.path}`, file.content)
-          } else if (file.type === 'dir') {
-            // Is Folder
-            fs.mkdirSync(`${pathStore}${file.path}`, { recursive: true })
-          }
-        })
-        em.emit('download-stop')
-        resolve(true)
-      })
+    const pathStore = `${process.cwd()}/ipfs-data/` // Path to store new ipfs-data
+
+    console.log(`files[0]: ${JSON.stringify(files[0])}`)
+    // console.log(`files[1]: ${JSON.stringify(files[1])}`)
+    // console.log(`files[2]: ${JSON.stringify(files[2])}`)
+
+    // CT: the file.type property is empty, so I'm not sure how to tell the
+    // difference between a file and a directory.
+    files.forEach(async (file, index) => {
+      // console.log(`index: ${index}`)
+      // console.log(`file.type: ${file.type}`)
+
+      // if (index === 0) {
+      //   // Is Folder
+      //   fs.mkdirSync(`${pathStore}${file.path}`, { recursive: true })
+      // } else {
+      //   // Is File
+      //   fs.writeFileSync(`${pathStore}${file.path}`, file.content)
+      // }
+
+      // Map files
+      // if (file.type === 'file') {
+      //   // Is File
+      //   fs.writeFileSync(`${pathStore}${file.path}`, file.content)
+      // } else if (file.type === 'dir') {
+      //   // Is Folder
+      //   fs.mkdirSync(`${pathStore}${file.path}`, { recursive: true })
+      // }
     })
+
+    return true
+
+    // Get the latest content from the IPFS network.
+    // return new Promise((resolve, reject) => {
+    //   console.log(`ping01`)
+
+    // ipfsNode.get(hash, async function (err, files) {
+    //   console.log(`ping02`)
+    //   if (err) {
+    //     console.error(`Error downloading files: `, err)
+    //     em.emit('download-stop')
+    //     reject(err)
+    //   }
+    //
+    //   console.log(`ping03`)
+    //
+    // const pathStore = `${process.cwd()}/ipfs-data/` // Path to store new ipfs-data
+    //
+    // files.forEach(async file => {
+    //   // Map files
+    //   if (file.type === 'file') {
+    //     // Is File
+    //     fs.writeFileSync(`${pathStore}${file.path}`, file.content)
+    //   } else if (file.type === 'dir') {
+    //     // Is Folder
+    //     fs.mkdirSync(`${pathStore}${file.path}`, { recursive: true })
+    //   }
+    // })
+    //   resolve(files)
+    //
+    //   // files.forEach(async file => {
+    //   //   // Map files
+    //   //   console.log(file)
+    //   //   if (file.type === 'file') {
+    //   //     // Is File
+    //   //     fs.writeFileSync(`${pathStore}${file.path}`, file.content)
+    //   //   } else if (file.type === 'dir') {
+    //   //     // Is Folder
+    //   //     fs.mkdirSync(`${pathStore}${file.path}`, { recursive: true })
+    //   //   }
+    //   // })
+    //
+    //   console.log(`ping04`)
+    //   em.emit('download-stop')
+    //   resolve(true)
+    // })
+    // })
   } catch (err) {
     console.error(`Error in ipfs.js/getContent()`)
     throw err
```

Comment:
These are the main changes. I think js-ipfs removed callback functionality of [`ipfs.get()`](https://github.com/ipfs/js-ipfs/blob/master/packages/interface-ipfs-core/SPEC/FILES.md#get). The code I have here is adapted from a [this unit test](https://github.com/ipfs/js-ipfs/blob/master/packages/interface-ipfs-core/src/get.js#L110-L115).

Reply:
Hi @christroutner ! here is my PR fixing the issues generated with the new update of IPFS https://github.com/Permissionless-Software-Foundation/ipfs-web-server/pull/9